.travis.yml

```diff
@@ -40,19 +47,26 @@ script:
     set -e
     if [ "$TRAVIS_BRANCH" == "gh-pages" ]
     then
-      timeout 120s htmlproofer --http-status-ignore 405,999 --url-ignore "/.*localhost.*/","/.*vimeo\.com.*/" --file-ignore "/.*\/files\/.*/"
+      make check-links-gh-pages
     else
-      if [ "$TRAVIS_PULL_REQUEST" == "false" -a "$TRAVIS_BRANCH" == "master" ]
+      if [ "$CHECK" == "htmlproofer" ]
+      then
+        make check-html
+      elif [ "$CHECK" == "yamllint" ]
+      then
+        make check-yaml
+      elif [ "$CHECK" == "slides" ]
+      then
+        make check-slides
+      elif [ "$CHECK" == "site-build" ]
       then
         make build
-      else
-        make check
       fi
     fi
 
 after_success:
   - |
-    if [ "$TRAVIS_PULL_REQUEST" == "false" -a "$TRAVIS_BRANCH" == "master" ]
+    if [ "$TRAVIS_PULL_REQUEST" == "false" -a "$TRAVIS_BRANCH" == "master" -a "$CHECK" == "site-build" ]
```

Comment:
It seems to me that this means that a commit on `master` or a merged PR will always publish the website even if the some checks failed.

Reply:
I added this because I thought otherwise each of the 4 jobs would try to do the publishing ..we don't want to skip publishing if links are broken I think (sometimes, like now, a third-party site may be down, and we still merge things and don't want this to stop updating of our own site) ..I figured as long as the site building job succeeded it was safe to update the site?

Reply:
If you want to publish only if the 3 check jobs pass, you can use build stages.
If you want to publish even if check jobs fail, then this is fine, but it's risky, you can publish a completely broken site.

Reply:
If it is totally broken, it will failed in `bin/publish` where it is checked that at least `index.html` is there

Reply:
@nsoranzo but don't we know that the site building itself at least succeeded because I'm having the job that ran `make build` do the publishing with an `after_success` ..or am I misunderstanding?

Reply:
Maybe I exaggerated, but you may surely push broken yaml files, no?
And HTML with broken internal and external links and hrefs.

Reply:
that is true (but if they are very broken it would fail the build too). But I could also add yaml checking to the `make build` command as well, to make sure both are in order before publishing?

Reply:
yep @ broken links, but for instance the site for quast has been down for the past two days orso now (but probably a temporary problem), and I don't know if we want to prevent ourselves from merging PRs and updating the site at such times (but this can be discussed) 

I do see that htmlproofer has options `--disable-external` and `--check-external` to check only internal or external links respectively, so we could consider adjusting things to at least require internal links to be working perhaps too